File: exercise/src/main/java/com/boo/exercise/MainActivity.java
package com.boo.exercise;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.viewpager.widget.ViewPager;

import com.boo.exercise.adapter.ViewPagerAdapter;
import com.boo.exercise.fragment.OneFragment;
import com.boo.exercise.fragment.ThreeFragment;
import com.boo.exercise.fragment.TwoFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        initViewPager();
    }

    /**
     * 初始化ViewPager页面滑动
     */
    private void initViewPager() {
        ViewPager viewPager = findViewById(R.id.viewPager);
        ViewPagerAdapter adapter = new ViewPagerAdapter(getSupportFragmentManager());
        adapter.addFragment(new OneFragment());
        adapter.addFragment(new TwoFragment());
        adapter.addFragment(new ThreeFragment());
        viewPager.setAdapter(adapter);
        viewPager.setCurrentItem(1);
    }
}

File: leetcode/src/main/java/com/boo/leetcode/MainActivity.kt
package com.boo.leetcode

import android.os.Bundle
import android.util.Log
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.boo.leetcode.utils.DefaultUtils

class MainActivity : AppCompatActivity() {
    companion object {
        private const val TAG = "MainActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    override fun onStart() {
        super.onStart()

        val height = intArrayOf(65, 70, 56, 75, 60, 68)
        val weight = intArrayOf(100, 150, 90, 190, 95, 110)

        val height1 = intArrayOf(175, 175, 175, 175, 175, 175)
        val weight1 = intArrayOf(140, 110, 105, 175, 125, 200)

        val a = DefaultUtils.bestSeqAtIndex(height1, weight1)
        Log.d(TAG, "onStart: $a")
    }
}